File: tests/py/test_standard.py
#!/usr/bin/env python3
"""Test entry point for standard message tests (Python)."""

import sys
import os

# Add include directory to path
sys.path.insert(0, os.path.join(os.path.dirname(__file__), 'include'))
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', 'generated', 'py'))

import standard_messages
from struct_frame.generated.serialization_test import get_message_info
from test_harness import run

TEST_NAME = "StandardMessages"
PROFILES = "standard, sensor, ipc, bulk, network"

if __name__ == "__main__":
    sys.exit(run(standard_messages, get_message_info, TEST_NAME, PROFILES))

File: tests/cpp/test_standard.cpp
#include "include/standard_messages.hpp"
#include "include/test_harness.hpp"

static constexpr char TEST_NAME[] = "StandardMessages";
static constexpr char PROFILES[] = "standard, sensor, ipc, bulk, network";

int main(int argc, char* argv[]) {
  return TestHarness<StandardMessages, structframe::serialization_test::get_message_info, TEST_NAME, PROFILES>::run(argc, argv);
}
